pkg/basicstation/cups/messages.go

```diff
@@ -84,6 +90,17 @@ func TokenCredentials(trust *x509.Certificate, authorization string) ([]byte, er
 	out = append(out, trust.Raw...)
 	// TODO: Refactor when client side TLS is supported https://github.com/TheThingsNetwork/lorawan-stack/issues/137
 	out = append(out, emptyClientCert...)
+	token := authorization
+	if strings.HasPrefix(authorization, "Bearer") {
+		t := strings.Split(authorization, " ")
+		if len(t) != 2 {
+			return nil, errInvalidAuthToken.WithAttributes("token", authorization)
+		}
+		token = t[1]
+	}
+	if _, _, _, err := auth.SplitToken(token); err != nil || len(token) != apiKeyLength {
+		return nil, errInvalidAuthToken.WithAttributes("token", authorization)
+	}
 	out = append(out, []byte(fmt.Sprintf("%s%s%s", "Authorization: ", authorization, "\r\n"))...)
```

Comment:
To prevent mistakes, why don't we do `strings.TrimRight(authorization, "\r\n")` here? That way we can be sure that we never add any extra CR or LF characters.